controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -167,8 +164,42 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		}
 
 		// TODO: Phases do not have checks at the moment, in order to make them more predictable and less error-prone, at the expense of increased computation cost.
-		// Creation phase.
-		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimCreating
+		// Validation phase.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimValidating
+
+		// If a StorageClass already exists:
+		// 	StorageClassClaim passes validation and is promoted to the configuring phase if:
+		//  * the StorageClassClaim has the same type as the StorageClass.
+		// 	* the StorageClassClaim has no encryption method specified when the type is filesystem.
+		// 	* the StorageClassClaim has a blockpool type and:
+		// 		 * the StorageClassClaim has an encryption method specified.
+		// 	  * the StorageClassClaim has the same encryption method as the StorageClass.
+		// 	StorageClassClaim fails validation and falls back to a failed phase indefinitely (no reconciliation happens).
+		existing := &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: r.storageClassClaim.Name,
+			},
+		}
+		if err = r.get(existing); err == nil {
+			sccType := r.storageClassClaim.Spec.Type
+			sccEncryptionMethod := r.storageClassClaim.Spec.EncryptionMethod
+			_, scIsFSType := existing.Parameters["fsName"]
+			scEncryptionMethod, scHasEncryptionMethod := existing.Parameters["encryptionMethod"]
+			switch {
+			case sccType == "filesystem" && scIsFSType && !scHasEncryptionMethod:
+			case sccType == "blockpool" && !scIsFSType && scHasEncryptionMethod && sccEncryptionMethod == scEncryptionMethod:
+			default:
+				r.log.Error(fmt.Errorf("storageClassClaim is not compatible with existing StorageClass"),
+					"StorageClassClaim validation failed.")
+				r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimFailed
+				return reconcile.Result{}, nil
+			}
```

Comment:
I think you are using the wrong tool from the language's toolbox.

A switch statement is a tool that solves the problem of implementation selection based on categorization into groups, where the group selection itself is based on a value falling within a specific range for each case.

An if statement is a logical switch to enable the selection of 1 of 2 code paths based on a boolean condition.

They have similarities but are not exactly the same. 

Usually when selecting the wrong tool the resulted code is harder to reason about, more prune to errors, and impared from a readability point of view

The intent behind this code is to implemet a decision point about the validity of the CR, to me this calls for an `if` statment. And you can even notice it in the code itself by identifing that all of you cases are fallthrough cases.

```suggestion
			if (sccType == "filesystem" && scIsFSType && !scHasEncryptionMethod) ||
				(case sccType == "blockpool" && !scIsFSType && scHasEncryptionMethod && sccEncryptionMethod == scEncryptionMethod) {
				r.log.Error(fmt.Errorf("storageClassClaim is not compatible with existing StorageClass"),
					"StorageClassClaim validation failed.")
				r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimFailed
				return reconcile.Result{}, nil
			}
```

Reply:
ACK